Rogue-Ape-Studios/Crusader
# 🍱 Revive item

## Content

Created an item that allows the player to gain a single revive, saving them from death **once**.

## Changes

### Added
`Assets/Prefabs/Items/ReviveItem.asset` : Was created with the prerequiste data for the new item.
`Assets/Scripts/Items/Core/Conditional/ReviveItem.cs` : Was created as a template for the ReviveItem.

### Updated
`Assets/Prefabs/Items/ItemManager.prefab` : Was updated with the new item.
`Assets/Scripts/Health/PlayerHealth.cs` : Was updated to support revive logic.
`Assets/Scripts/Items/Manager/ItemManager.cs` : Was updated with fixes for `NullReferenceExceptions` and minor code improvements.

## Testing

The feature / Bugfix can be testing by following these steps:

1. Load up a scene.
2. Make sure the `EventHolder` in `ItemManager` has the correct events (branch will be created for that soon).
3. Play the game.
4. Once a legendary pickup spawns, pick up the item.
5. Die.
6. Get revived.

Alternatively, you can toggle `hasRevive` in the inspector under `PlayerHealth`.

Aids #340
